fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1347,8 +1347,9 @@ private FDBRawRecord reconstructSingleRecord(final Subspace recordSubspace, fina
         final RecordCursor<FDBRawRecord> rawRecords;
         if (metaData.isSplitLongRecords()) {
             // Note that we always set "reverse" to false since regardless of the index scan direction, the MappedKeyValue is in non-reverse order
+            // Setting the limit manager to UNTRACKED since the KeyValueCursor already counts records, and we don't want the unsplitter to double-count
             rawRecords = new SplitHelper.KeyValueUnsplitter(context, recordSubspace, rangeCursor, oldVersionFormat,
-                    sizeInfo, false, new CursorLimitManager(scanProperties));
+                    sizeInfo, false, CursorLimitManager.UNTRACKED);
```

Comment:
Does this need to do the same below (in the case of `!metaData.isSplitLongRecords` but where `omitUnsplitSuffix` is `false`)?

Reply:
Clearly an issue. Trying to figure out why the test that was supposed to catch that (RemoteFetchWithoutSplitRecordsTest) didn't.

Reply:
Test didn't pick this up because we are still in api version `6.3`by default. Locally running with 7.1 does fail.
Should we change the default version to 7.1 to reduce these kinds of issues (as we should be moving to supporting this version across the board)?